pkg/network/encoding/marshal/usm.go

```diff
@@ -93,13 +93,18 @@ func GroupByConnection[K comparable, V any](protocol string, data map[K]V, keyGe
 // Find returns a `USMConnectionData` object associated to given `network.ConnectionStats`
 // The returned object will include all USM aggregation associated to this connection
 func (bc *USMConnectionIndex[K, V]) Find(c network.ConnectionStats) *USMConnectionData[K, V] {
+	if c.Type != network.TCP {
+		return nil
+	}
+
 	result := bc.find(c)
 	if result != nil {
 		// Mark `USMConnectionData` as claimed for the purposes of orphan
 		// aggregation reporting
 		result.claimed = true
 	}
 
+	log.TraceFunc(func() string { return fmt.Sprintf("could not find connection %+v in usm data", c) })
```

Comment:
please wrap with log.ShouldLog, as it will still allocate memory for every call

Reply:
source for the claim - https://github.com/DataDog/datadog-agent/pull/17354

Reply:
This is not really relevant to the PR, but using `ShouldLog` really clutters the code. I would rather take the middle path here and use `TraceFunc`, which is still clunky, but at least a little better than using `ShouldLog`; it also allocates way less per call than just `Trace`. I'll change these, but I'm going to keep the ones in `state.go`; I haven't seen these show up in memory/cpu profiles in my testing.